Python/flowgraph.c

```diff
@@ -1385,15 +1385,24 @@ optimize_basic_block(PyObject *const_cache, basicblock *bb, PyObject *consts)
                         }
                         cfg_instr *is_instr = &bb->b_instr[i + 1];
                         cfg_instr *jump_instr = &bb->b_instr[i + 2];
+                        cfg_instr *not_instr = NULL;
                         // Get rid of TO_BOOL regardless:
                         if (jump_instr->i_opcode == TO_BOOL) {
                             INSTR_SET_OP0(jump_instr, NOP);
                             if (bb->b_iused <= i + 3) {
                                 break;
                             }
-                            jump_instr = &bb->b_instr[i + 3];
+                            jump_instr++;
+                        }
+                        bool invert = false;
+                        if (jump_instr->i_opcode == UNARY_NOT) {
+                            if (bb->b_iused <= i + 4) {
+                                break;
+                            }
+                            not_instr = jump_instr;
+                            jump_instr++;
+                            invert = true;
                         }
```

Comment:
See my comments above. I think these blocks need to be flipped, since `UNARY_NOT` comes before `TO_BOOL`.

The instructions emitted by the code generator on this branch are:
- `if x is None`/`if not x is not None`: `LOAD_* LOAD_CONST IS_OP TO_BOOL POP_JUMP_IF_*`
- `if x is not None`/`if not x is None`: `LOAD_* LOAD_CONST IS_OP UNARY_NOT TO_BOOL POP_JUMP_IF_*`